### DISCUSSION TITLE
The trait `Iden` is not implemented for `&str`

### DISCUSSION THREAD

**Discussion by mamcx** [SeaQL/sea-query#154] [Q&A] [answered]
Any reason is required to use enums +ident for tables/fields instead of passing a simple string?

```rust
impl ZoneRow {
    fn query() -> SelectStatement {
        Query::select().from("zone").to_owned() //not work
    }
}
```

**Comment by tyt2y3** [comment, marked as answer]
If you need to, you can do Alias::new("zone").

Or you can 
#[derive(Iden)]
struct Zone;

There are 2 reasons:

1) Correctness. it's an identifier, so if you misspelt it, it'd be a compile error

2) Efficiency. &str is bad, because the syntax tree needs to store the name *until* you finally serialize it. Internally it must make a clone of the string (or end up in a lifetime struggle I guess).
But now, the identifier is a pointer, so there is only one unique copy of your string in memory. &'static str would also be great, but then it limits us from runtime behaviour.

**Comment by mamcx** [comment]
Number 1 is still a little unexpected because I was under the impression SeaSQL was for runtime made query building, but I can derive my own Iden easily:

```rust
struct RawName {
    named:String
}

impl  Iden for RawName {
    fn unquoted(&self, s: &mut dyn Write) {
        write!(
            s,
            "{}",
            &self.named
        )
            .unwrap();
    }
}
```

Number 2 makes sense, but this still requiere to return owned query:

```rust
impl ZoneRow {
    fn query() -> &'static mut SelectStatement {
        Query::select().from(Tables::Zone) <-- error[E0515]: cannot return value referencing temporary value
    }
}
```

Is not a big deal for me, returning SelectStatement is similar to how my own ORM is working now.

I close this because exist acceptable workarounds.

**Comment by tyt2y3** [comment]
Thank you for your understanding.
